File: instrumentation/play-ws-2.13_2.6.0/src/main/scala/com/nr/agent/instrumentation/playws/OutboundWrapper.scala
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.agent.instrumentation.playws

import com.newrelic.api.agent.{HeaderType, OutboundHeaders}

import scala.collection.mutable

class OutboundWrapper(var nrHeaders: mutable.Buffer[(String, String)]) extends OutboundHeaders {

  override def getHeaderType: HeaderType = HeaderType.HTTP

  override def setHeader(name: String, value: String): Unit = {
    nrHeaders += (name -> value)
  }

}

File: instrumentation/httpclient-5.0/src/main/java/com/nr/agent/instrumentation/httpclient50/OutboundWrapper.java
/*
 *
 *  * Copyright 2023 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.agent.instrumentation.httpclient50;

import com.newrelic.api.agent.HeaderType;
import com.newrelic.api.agent.OutboundHeaders;
import org.apache.hc.core5.http.HttpRequest;

public class OutboundWrapper implements OutboundHeaders {

    private final HttpRequest delegate;

    public OutboundWrapper(HttpRequest request) {
        this.delegate = request;
    }

    @Override
    public void setHeader(String name, String value) {
        delegate.setHeader(name, value);
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }
}
